examples/v1beta1/pipelineruns/alpha/pipelinerun-with-matrix.yaml

```
@@ -51,3 +51,8 @@ spec:
             value: chrome
         taskRef:
           name: platform-browsers
+      - name: matrix-with-skipped-tasks # matrix parameters cannot have empty array values
+        matrix:
+          params:
+            - name: platform
+              value: '[]'
```

this is not a fully formed pipelinetask - needs a taskspec or taskref

added a taskspec, and implemented it such that this task should fail if it's executed (to ensure that it is always skipped)
```suggestion
      - name: matrix-params-with-empty-array-skipped
        matrix:
          params:
            - name: versions
              value: []
        taskSpec:
          params:
            - name: version
          steps:
            - name: echo
              image: ubuntu
              script: exit 1
```

also, in another PR, let's add the same examples under finally 🙏🏾